exercises/concept/wizards-and-warriors-2/.meta/src/reference/java/GameMaster.java

```diff
@@ -0,0 +1,28 @@
+public class GameMaster {
+
+    public static String describe(Character character) {
+        return "You're a level " + character.getLevel() + " " + character.getCharacterClass() + " with " +
+               character.getHitPoints() + " hit points.";
+    }
+
+    public static String describe(Destination destination) {
+        return "You've arrived at " + destination.getName() + ", which has " + destination.getInhabitants() + " " +
+               "inhabitants.";
+    }
+
+    public static String describe(TravelMethod travelMethod) {
+        if (travelMethod == TravelMethod.WALKING) {
+            return "You're traveling to your destination by walking.";
+        }
+        return "You're traveling to your destination on horseback.";
+
+    }
+
+    public static String describe(Character character, Destination destination, TravelMethod travelMethod) {
+        return describe(character) + " " + describe(travelMethod) + " " + describe(destination);
+    }
+
+    public static String describe(Character character, Destination destination) {
+        return describe(character) + " " + describe(TravelMethod.WALKING) + " " + describe(destination);
```

Comment:
Actually maybe it would be good practice for students to learn calling one overload from another as part of this exercise, since that is very common in Java, especially when using multiple constructors.

```suggestion
        return describe(character, destination, TravelMethod.WALKING);
```